hystrix-core/src/main/java/com/netflix/hystrix/strategy/HystrixArchaiusHelper.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.hystrix.strategy;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import com.netflix.hystrix.strategy.properties.HystrixDynamicProperties;
+
+/**
+ * @ExcludeFromJavadoc
+ * @author agentgt
+ */
+class HystrixArchaiusHelper {
+
+    /**
+     * To keep class loading minimal for those that have archaius in the classpath but choose not to use it.
+     * @ExcludeFromJavadoc
+     * @author agent
+     */
+    private static class LazyHolder {
+        private final static Method loadCascadedPropertiesFromResources;
+        private final static String CONFIG_MANAGER_CLASS = "com.netflix.config.ConfigurationManager";
+    
+        static {
+            Method load = null;
+            try {
+                Class<?> configManager = Class.forName(CONFIG_MANAGER_CLASS);
+                load = configManager.getMethod("loadCascadedPropertiesFromResources", String.class);
+            } catch (Exception e) {
+            }
+    
+            loadCascadedPropertiesFromResources = load;
+        }
+    }
+
+    /**
+     * @ExcludeFromJavadoc
+     */
+    static boolean isArchaiusV1Available() {
+        return LazyHolder.loadCascadedPropertiesFromResources != null;
+    }
+
+    static void loadCascadedPropertiesFromResources(String name) {
+        if (isArchaiusV1Available()) {
+            try {
+                LazyHolder.loadCascadedPropertiesFromResources.invoke(null, name);
```

Comment:
Basically this code comes from the original 1.4 code https://github.com/Netflix/Hystrix/blob/1.4.x/hystrix-core/src/main/java/com/netflix/hystrix/strategy/HystrixPlugins.java#L57

What it does is indirectly boots up Archaius and adds the hystrix-plugins properties file as an additional configuration resource. The actual method does not return anything.The assumption here is Archaius isn't going to provide configuration properties to point to use a different `HystrixDynamicProperties`.

This is only to create `HystrixDynamicProperties`... all the other implementations will be loaded using cascaded configuration (assuming you have Archaius). That is `hystrix-plugins.properties` will be used.

This does bring up an excellent point though that the `HystrixDynamicPropertiesSystemProperties` (I picked that name to be consistent with the naming scheme) perhaps should load `hystrix-plugins.properties`? Let me know what you think about that.